langchain/src/vectorstores/supabase.ts

```diff
@@ -0,0 +1,107 @@
+import { SupabaseClient } from '@supabase/supabase-js';
+import { VectorStore } from "./base.js";
+import { Embeddings } from "../embeddings/base.js";
+import { Document } from "../document.js";
+
+type SupabaseMetadata = Record<string, any>;
+
+export type SearchEmbeddingsResponse = {
+    id: number
+    prompt: string
+    completion: string
+    similarity: number
+    token_count: number
+  }
+
+  export type SearchEmbeddingsParams = {
+    query_embedding: number[]
+    match_threshold: number // float
+    match_count: number // int
+  }
+  
+  
+export class PGVectorStore extends VectorStore {
```

Comment:
I guess the name should refer to Supabase, given it requires a supabase client

Reply:
Yeah. theres also some other serverless pg providers, my original thought was it could be generalized